sdk/clientcore/core/src/main/java/io/clientcore/core/instrumentation/logging/ExceptionLoggingEvent.java

```diff
@@ -0,0 +1,117 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package io.clientcore.core.instrumentation.logging;
+
+import io.clientcore.core.instrumentation.InstrumentationContext;
+
+import java.util.function.BiFunction;
+
+/**
+ * This class is used to create a logging event for exception with contextual
+ * information that's consistently populated in exception and log message.
+ *
+ * @param <T> The type of the exception to be logged.
+ */
+public class ExceptionLoggingEvent<T extends Throwable> {
```

Comment:
Does this need to be a different type than `LoggingEvent`?

Did this solve the issue with Spotbugs where all usages of log with throwing causing unchecked cast warnings from Spotbugs?

Reply:
If keeping this add `@Metadata(properties = FLUENT)`

Reply:
yes, we need a new type for a few reasons:
1. we need `log` to return exception for ergonomics reasons. An alternative would be to make `LoggingEvent` a template and having a regular one return `Void`
2. Exception logging event is special - it's always functional and always records key-value-pairs since you create it with intent to throw exception unrelated to log levels.  Therefore it does not need a bunch of lazy/supplier features we have on the LoggingEvent.